index.js

```diff
@@ -104,128 +122,178 @@ function openbabelFileConversion(req, res, outputName, options = [], inputFileTy
   
   let responseIsSent = false;
   let form = new formidable.IncomingForm();
-  
-  form.multiples = true;
-  form.parse(req);
-  form.on('field', function(name, value) {
-    fields[name] = value
-  })
-  
+
   //create a random name for the directory
   let nameHash = crypto.createHmac('sha1', crypto.randomBytes(48))
     .update(Date.now()
     .toString())
     .digest('hex');
-    
-  let directoryPath = path.join(__dirname,'uploads', nameHash)
-  let uploadsPath = path.join(__dirname,'uploads')
+
+  let jobPath = path.join(nameHash+"/")
+  let directoryPath = path.join(uploadsPath, jobPath)
   
-  //Check for naming collisions
-  if (!fs.existsSync(directoryPath)) {
-    //create the new directory
+  // Ensure there is no hash collision
+  checkExists(jobPath, (err, exists) => {
+    if (err || exists) {
+      if(!responseIsSent) {
+        res.status(500);
+        res.send("Storage Error. Please try again.");
+        responseIsSent = true;
+      }
+    }
+    // Make the storage space for the job
     fs.mkdirSync(directoryPath);
-  }
-  else if(!responseIsSent) {
-    res.status(500);
-    res.send("Storage Error. Please try again.");
-    responseIsSent = true;
-  }
-  
-  form.on('fileBegin', function (name, file){
-    //each field in the form whose name starts with "molecule"
-    if (name.startsWith('molecule')) {
-      molecule = {}
-      molecule.name = file.name
-      molecules.push(molecule)
+    if (useCloudStorage) {
+      storage.bucket(bucket).file(jobPath).save('', {resumable:false}, (err) => {
+        if (err) {
+          console.log(err);
+          if(!responseIsSent) {
+            res.status(500)
+            res.send('Error storing result')
+            responseIsSent = true;
+          } 
+        }
+      })
     }
 
-    fullPath = path.join(directoryPath, file.name);
-    file.path = fullPath;
-  });
-  form.on('end', function() {
-    try {
-      //arguments for openbabel
-      let args = []
-
-      let execOptions = [] 
+    form.multiples = true;
+    form.parse(req);
+    form.on('field', function(name, value) {
+      fields[name] = value
```

Comment:
You don't define fields anywhere, probably should explicitly specify that with a let

Reply:
It's defined with a `lets` on line 121.